src/config.c

```diff
@@ -2400,6 +2401,17 @@ static int isValidProcTitleTemplate(char *val, const char **err) {
     return 1;
 }
 
+/* Validate specified string is a valid locale string, meanwhile do
+ * the setting work. */
+static int isValidLocale(char *val, const char **err) {
+    const char *s = setlocale(LC_COLLATE, val);
```

Comment:
this should be done in the "apply" callback, not the "is valid" callback.
i.e. just rename the function and change the order of arguments in the createStringConfig line.

additionally, this callback isn't called on startup (when parsing a config file). so the locale from server.locale, should be applied in `initServer()`.
i.e. just move the existing call in server.c to be inside `initServer()` and use `server.locale` instead of the hard coded `""`.

Reply:
Thanks for the correction. I'll applied the changes.

Reply:
Done